marlowe-runtime/proxy-api/Language/Marlowe/Protocol/Server.hs

```diff
@@ -33,6 +45,15 @@ data MarloweRuntimeServer m a = MarloweRuntimeServer
   , recvMsgRunContractQuery :: m (PeerTraced (Query ContractRequest) 'AsServer 'Query.StReq m a)
   } deriving Functor
 
+data MarloweRuntimeServerDirect m a = MarloweRuntimeServerDirect
```

Comment:
Added a new flavour of server when in-process communication is used. This could be used for interprocess communication as well, but it would be far less efficient.